sympy/solvers/bivariate.py

```diff
@@ -196,6 +198,50 @@ def _solve_lambert(f, symbol, gens):
       a = -1, d = a*log(p), f = -log(d) - g*log(p)
     """
 
+    def _solve_even_degree_expr(expr, symbol):
+        """
+        Helper routine to replace ``t`` with ``+/- symbol`` in ``expr`` to
+        return a list of solutions from the two ``expr`` generated.
+
+        For example, if ``expr = 2*log(t) + x/2` then solutions for
+        ``2*log(x) + x/2 = 0`` and ``2*log(-x) + x/2 = 0`` would be returned.
```

Comment:
```
if ``expr = 2*log(t) + x/2`` then solutions form ``2*log(x) + x/2 = 0`` and ``2*log(-x) + x/2 = 0``
```

The above statement is mathematically incorrect. The solution for the expression on the left cannot be derived from the two expressions on the right. The reason being:

The domain of
- `2*log(t) + x/2` and
- `log(t**2) + x/2`
 are not same. You rather want to replace the equation on the left with `log(t**2) + x/2` as the domain of this equation is all reals (positive and negative) except 0, but for the former it's only the positive reals.

Reply:
That's what was throwing me off, too. Solving a bivariate like this is strange. We are not trying to solve the log equation, we are trying to solve the equation from which the log equation came: `x**2*exp(x/2) - 1` and for that equation solutions of `2*log(x) + x/2` and `2*log(-x) + x/2` give all three solutions of that equation. The sentence that you are referring to is a statement of fact -- this is what the helper function is doing -- while the paragraph after it is the explanation of *why* it is doing that.

And I'm not sure anything after the paragraph is necessary; it is redundant given what comes before.

Reply:
@aktech `2*log(t) + x/2` this is written beacuse the function recieves this type of expression. We are describing the functionality of `_solve_even_degree_expr`. 
`log(t**2) + x/2` this expression is meaningless for this function because when t is replaced with +/-x then eventually the equation will become `log(x**2) + x/2`

> as the domain of this equation is all reals (positive and negative) except 0, but for the former it's only the positive reals.

This thing is already considered before entering this function(_solve_even_degree_expr)

Reply:
I think we should write that 'for the equation x**2*exp(x/2) - 1 the expr will be 2*log(t) + x/2 in which the the t will be replaced with +/-x.'